baybe/recommenders/pure/bayesian/botorch.py

```diff
@@ -76,6 +85,13 @@ class BotorchRecommender(BayesianRecommender):
     optimization. **Does not affect purely discrete optimization**.
     """
 
+    max_n_subspaces: int = field(default=10, validator=[instance_of(int), ge(1)])
```

Comment:
@AVHopp @Scienfitz: To you think this is an appropriate attribute name?

Reply:
I think that it should be linked to `cardinality` somehow: For people that are not interested in cardinality constraints, it is only clear after reading the docstring that this is not interesting for them. This would however make the name quite long :/  So although I am not perfectly happy with the name, we could keep it as I do not see a better alternative while keeping it here.

Reply:
I get your point but I see this emerging as part of a bigger problem, actually, not limited to this particular attribute. Namely that all attributes of the class are linked to particular settings only:
* `sequential_continuous` is relevant for continuous spaces only
* `hybrid_samples` is relevant for hybrid spaces only
* `sampling_percentage` is relevant for hybrid spaces and specific sampling strategies only

In that sense, `max_n_subspaces` is even a little less problematic since it is strictly speaking not only relevant for cardinality constraints but could also be used for hybrid spaces. And now I also see another clash, since it's functionality basically overlaps with `sampling_percentage` 🤔 

So two TODOs from I see:
* Cleverly resolve the duplication somehow
* Potentially, in a next PR: find a better structure. We could consider, for example, to capture all the settings in a separate object like a TypedDict or so, that then uses its own validation path

@Scienfitz: Thoughts from your end?

Reply:
To avoid the duplicates, we could rename `sampling_percentage` to `hybrid_sampling_percentage` or something similar, that might already be sufficient. Agree that it would be beneficial to somehow "merge" these two attributes soon.